fearless/Modules/Staking/Services/StakingServiceFactory.swift

```diff
@@ -0,0 +1,81 @@
+import Foundation
+import RobinHood
+
+protocol StakingServiceFactoryProtocol {
+    func createEraValidatorService(for chainId: ChainModel.Id) throws -> EraValidatorServiceProtocol
+    func createRewardCalculatorService(
+        for chainId: ChainModel.Id,
+        assetPrecision: Int16,
+        validatorService: EraValidatorServiceProtocol
+    ) throws -> RewardCalculatorServiceProtocol
+}
+
+final class StakingServiceFactory: StakingServiceFactoryProtocol {
+    let chainRegisty: ChainRegistryProtocol
+    let storageFacade: StorageFacadeProtocol
+    let eventCenter: EventCenterProtocol
+    let operationManager: OperationManagerProtocol
+    let logger: LoggerProtocol?
+
+    private lazy var substrateDataProviderFactory = SubstrateDataProviderFactory(
+        facade: storageFacade,
+        operationManager: operationManager
+    )
+
+    init(
+        chainRegisty: ChainRegistryProtocol,
+        storageFacade: StorageFacadeProtocol,
+        eventCenter: EventCenterProtocol,
+        operationManager: OperationManagerProtocol,
+        logger: LoggerProtocol? = nil
```

Comment:
Because of logging errors I don't like how we declare logger to be Optional and give it `nil` default value. It is easy to forget to pass an actual logger instance and no logs will be produced. Can we discuss alternatives?